packages/http-client-csharp/generator/Microsoft.Generator.CSharp.ClientModel/src/Providers/MrwSerializationTypeDefinition.cs

```diff
@@ -442,13 +442,21 @@ internal MethodProvider BuildDeserializationMethod()
         private MethodBodyStatement[] BuildAbstractDeserializationMethodBody()
         {
             var unknownVariant = _model.DerivedModels.First(m => m.IsUnknownDiscriminatorModel);
+            bool onlyContainsUnknownDerivedModel = _model.DerivedModels.Count == 1;
+            IfStatement? deserializeDiscriminatedModelsConditions = null;
+
+            if (!onlyContainsUnknownDerivedModel)
```

Comment:
nit: we can just always initialize `deserializeDiscriminatedModelsCondition` and then check for `onlyContainsUnknownDerivedModel` on L459.